lib/trie/database.go

```diff
@@ -128,6 +128,26 @@ func (t *Trie) load(db chaindb.Database, curr node) error {
 	return nil
 }
 
+//GetDBKey get hash of each node.
+func (t Trie) GetDBKey(curr node, keys map[common.Hash]interface{}) error {
```

Comment:
if you're using the `keys` map just to check existence of a key, maybe change to `map[common.Hash]struct{}`